hilbertcurve/hilbertcurve.py

```diff
@@ -135,6 +145,13 @@ def distance_from_coordinates(self, x_in: List[int]) -> int:
             raise ValueError(
                 'invalid coordinate input x={}.  one or more dimensions have a '
                 'value less than 0'.format(x))
+        
+        if any(not elx % 1 for elx in x):
+            raise TypeError(
+                'invalid coordinate input x={}. one or more dimensions is not '
+                'an integer and can not be converted'
+                
+        for i in range(len(x)): x[i] = int(x[i])
```

Comment:
are you avoiding a list comprehension here to preserve the type of `x`? 
my first thought was why not use `x = [int(x) for el in x]` but then it will always be a list and it seems you were keen to accept any iterable of "convertible to ints"

Reply:
That is a much cleaner and more Pythonesque solution. Also, there is no need to preserve data type because x was already turned into a list at line 135